clippy_lints/src/types/mod.rs

```diff
@@ -167,8 +167,8 @@ declare_clippy_lint! {
     /// Check the [Box documentation](https://doc.rust-lang.org/std/boxed/index.html) for more information.
     ///
     /// ### Why is this bad?
-    /// Any `&Box<T>` can also be a `&T`, which is more
-    /// general.
+    /// A `&Box<T>` parameter requires the function caller to box `T` first before passing it to a function.
+    /// Using `&T` defines a concrete type for the parameter and generalizes the function.
```

Comment:
I think it should also include something like "a `&Box<T>` will auto-deref to `&T` at the function call site".

Reply:
Would something like this work:
A `&Box<T>` parameter will auto-deref to `&T` at the function call site and requires the function caller to box `T` first before passing it to a function. 
Using `&T` defines a concrete type for the parameter and generalizes the function.

Reply:
I mean that if the parameter type is changed to `&T`, then you may still have `&Box<T>` passed to the function and it will automatically deref to `&T` with having to add a deref operator. The coder may not be aware of this, and that is IMO a likely reason that they choose to keep the Box in the first place.

Reply:
My bad, in that case, it would be something like this:
A `&Box<T>` parameter requires the function caller to box `T` first before passing it to a function. 
Using `&T` defines a concrete type for the parameter and generalizes the function, this would also auto-deref to `&T` at the function call site if passed a `&Box<T>`.

Reply:
Ya!